theme/story.tsx

```diff
@@ -0,0 +1,50 @@
+import type { Culture, Story } from '@prezly/sdk';
+import type { ReactNode } from 'react';
+
+import { api } from '@/theme-kit';
+
+type Match = { slug: string } | { uuid: string };
+
+export async function match(params: Match) {
+    const { contentDelivery } = api();
+
+    const story = await contentDelivery.story(params);
```

Comment:
Continuing my comment in `gallery-album.tsx`, I don't get why we can't fetch the story in the Page component and call Next's `notFound()` method when the story is not found?

This would make the props logic much simpler and cleaner.

Reply:
> I don't get why we can't fetch the story in the Page component and call Next's notFound() method when the story is not found?

Yeah, you can do that in the `gallery-album` module. As there are guaranteed to be no other candidates for that URL.

But it is different here, for the `story` module -- the route is competing with the same-hierarchy-level `/[locale]` route, so we want to cut it off on the earlier stage: before the rendering happens.

Potentially, this can be changed here too, as the router [is locale-aware now](https://github.com/prezly/theme-nextjs-bea/pull/912#discussion_r1358216549) (I've added that logic on rather late stages), so story fetching can be moved to the rendering phase without having any unwanted consequences.

However, as I have already said in [my other comment](https://github.com/prezly/theme-nextjs-bea/pull/912#discussion_r1358233949): there is still a big difference compared to `getServerProps` approach -- as this is only fetching the direct route-bound data objects, allowing the components to fetch the rest using direct API interactions. 

So it's still much simpler. 

Anway, I'll think about this more. Thank you for this remark.